File: ScaleformUI_Csharp/Menus/Pause Menus/Elements/Items/FakeBlip.cs
using CitizenFX.Core;

namespace ScaleformUI.PauseMenus.Elements.Items
{
    public class FakeBlip
    {
        public int Sprite;
        public Vector3 Position;
        public float Scale = 0;
        public BlipColor Color = BlipColor.White;
        public FakeBlip() { }
        public FakeBlip(int sprite, Vector3 position)
        {
            Sprite = sprite;
            Position = position;
        }
        public FakeBlip(BlipSprite sprite, Vector3 position)
        {
            Sprite = (int)sprite;
            Position = position;
        }
    }
}

File: ScaleformUI_Lua/src/Menus/PauseMenus/LobbyMenu/Items/FakeBlip.lua
FakeBlip = setmetatable({}, FakeBlip)
FakeBlip.__index = FakeBlip
FakeBlip.__call = function()
    return "LobbyItem", "FakeBlip"
end

---@class FakeBlip
---@field private New fun(sprite:number, position:vector3, color:HudColours, scale:number)
---@field public Sprite number
---@field public Position vector3

function FakeBlip.New(sprite, position, color, scale)
    local _data = {
        Sprite = sprite,
        Position = position,
        Scale = scale or 0.0,
        Color = color or HudColours.HUD_COLOUR_WHITE
    }
    return setmetatable(_data, FakeBlip)
end
